xarray/core/groupby.py

```diff
@@ -209,6 +217,9 @@ def __getitem__(self, key):
             key = key[0]
         return self.values[key]
 
+    def copy(self, deep: bool = True, data: Any = None):
+        raise NotImplementedError
```

Comment:
I (and mypy) don't understand how `_DummyGroup` manages to get through the GroupBy init without crashing?